### ISSUE TITLE
Add to multiple servers

### ISSUE THREAD

**Issue by VerstreuteSeele** [zacimac/warnable#11]
Hey. Im having issues configuring this bot for multiple servers. can you maybe help me with that?

**Comment by VerstreuteSeele** [comment]
Im also getting an error when trying to run the bot on just a single server. "unexpected identifier "points"" in the config

**Comment by VerstreuteSeele** [comment]
Also.. even on a single server with fully filled out config, the bot doesnt do anything :/

**Comment by zacimac** [comment]
I believe there may be an issue with your config file. Please post it in here in a code block **(Remove the token first before sending!)**

**Comment by VerstreuteSeele** [comment]
will do in about an hour. currently not at home👍

**Comment by VerstreuteSeele** [comment]
// Warnable 2.0.0 - Config
module.exports = {
    token: "", // Discord Bot Token - Obtained from the Discord Developer Portal.

    prefix: "c!", // Prefix for commands.

    guilds: { // Guilds must be added in here for the bot to work in.
        "822728900693196800": { 
            point: [ // Actions for if a user reaches above any point values.
                // Range: The range of points for the action to trigger; Points must be lower number then higher number. 
                // Message: "%guild" = The server name / "%points" = Their total points value when the action took place.
                // Leave message blank ("") to disable.
                {
                    range: "5-7",
                    action: "mute-1d",
                    message: "You have been temporarily muted for one day in %guild"
                },
                {
                    range: "8-11",
                    action: "mute-7d",
                    message: "You have been temporarily muted for seven days in %guild"
                },
                {
                    range: "12-14",
                    action: "ban-7d",
                    message: "You have been temporarily banned for seven days in %guild"
                },
                {
                    range: "15-100",
                    action: "ban",
                    message: "You have been permanently banned in %guild for reaching %points"
                }
            ],
			 directmessage: "⚠ **You have been warned!** ⚠\n> You recieved %points point(s) for `%reason`.\n> You now have a total of %total point(s).", // Direct message the user whenever warned. "" to disable. 
            // "\n" = next line. "%points" = points warned, "%total" = their total points, "%reason" = the reason they were warned.
            roles: { // Role ID's for the guild.
                mute: "", // Used for the mute function.
                immune: [826123413327708200], // Auto mod ignores anyone with the role applied.
                moderator: [], // Moderators can only view warnings.
                admin: [826123413327708200] // Admins can apply, remove warnings and adjust guild settings.
            },
            channels: { // Channel ID's to send logs. "" to disable > Note: Logs will still display in the console if disabled.
                important: "", // Actions (mute, kick or ban) on a user from points or manual commands.
                warnings: "", // New warnings that were added to a user.
                messages: "", // Edits or deletions to messages.
                users: "" // Users join or leaving.
            }
			 
			}
		},
	

    msg: { // Bot message settings.
        // Colors used for embeds (https://zachary.fun/i/0vk7L.png)
        colorSuccess: 0x2ecc71,
        colorError: 0xe74c3c
    }
}



I wasnt too sure with where to add the guild but I tried a few different ways. heres the one which didnt give me an error in the console

pls let me know whats wrong and how to add more guilds

**Comment by VerstreuteSeele** [comment]
After re-downloading the bot and filling in the config once more I came a step further. Now the bot recognizes my commands **however** now I´m getting "You do not have permission to use that command." Even tho I double checked and my RoleID is correctly in the config.

**Comment by zacimac** [comment]
> The config file: 
> ```
> // Warnable 2.0.0 - Config
> module.exports = {
>     token: "", // Discord Bot Token - Obtained from the Discord Developer Portal.
> 
>     prefix: "c!", // Prefix for commands.
> 
>     guilds: { // Guilds must be added in here for the bot to work in.
>         "822728900693196800": { 
>             point: [ // Actions for if a user reaches above any point values.
>                 // Range: The range of points for the action to trigger; Points must be lower number then higher number. 
>                 // Message: "%guild" = The server name / "%points" = Their total points value when the action took place.
>                 // Leave message blank ("") to disable.
>                 {
>                     range: "5-7",
>                     action: "mute-1d",
>                     message: "You have been temporarily muted for one day in %guild"
>                 },
>                 {
>                     range: "8-11",
>                     action: "mute-7d",
>                     message: "You have been temporarily muted for seven days in %guild"
>                 },
>                 {
>                     range: "12-14",
>                     action: "ban-7d",
>                     message: "You have been temporarily banned for seven days in %guild"
>                 },
>                 {
>                     range: "15-100",
>                     action: "ban",
>                     message: "You have been permanently banned in %guild for reaching %points"
>                 }
>             ],
> 			 directmessage: "⚠ **You have been warned!** ⚠\n> You recieved %points point(s) for `%reason`.\n> You now have a total of %total point(s).", // Direct message the user whenever warned. "" to disable. 
>             // "\n" = next line. "%points" = points warned, "%total" = their total points, "%reason" = the reason they were warned.
>             roles: { // Role ID's for the guild.
>                 mute: "", // Used for the mute function.
>                 immune: [826123413327708200], // Auto mod ignores anyone with the role applied.
>                 moderator: [], // Moderators can only view warnings.
>                 admin: [826123413327708200] // Admins can apply, remove warnings and adjust guild settings.
>             },
>             channels: { // Channel ID's to send logs. "" to disable > Note: Logs will still display in the console if disabled.
>                 important: "", // Actions (mute, kick or ban) on a user from points or manual commands.
>                 warnings: "", // New warnings that were added to a user.
>                 messages: "", // Edits or deletions to messages.
>                 users: "" // Users join or leaving.
>             }
> 			 
> 			}
> 		},
> 	
> 
>     msg: { // Bot message settings.
>         // Colors used for embeds (https://zachary.fun/i/0vk7L.png)
>         colorSuccess: 0x2ecc71,
>         colorError: 0xe74c3c
>     }
> }
> ```
> 
> 
> I wasnt too sure with where to add the guild but I tried a few different ways. heres the one which didnt give me an error in the console
> 
> pls let me know whats wrong and how to add more guilds

In this config, you have the "points" property as just "point" so it couldn't read it.

**Comment by zacimac** [comment]
> After re-downloading the bot and filling in the config once more I came a step further. Now the bot recognizes my commands **however** now I´m getting "You do not have permission to use that command." Even tho I double checked and my RoleID is correctly in the config. 

From looking at your previous config, you need to have the IDs surrounded in quotation marks (").

### Example:
For single roles: `["idhere"]`
For multiple roles: `["roleid1", "roleid2"]`

Any that don't have the square brackets are just single roles only, make sure they are still surrounded with quotation marks.

**Comment by VerstreuteSeele** [comment]
Thanks. now it works

How do I add another server to the config tho? and is there maybe a way to do that with a command instead of editing the config?

A way for this would be for the bot to check if you have the "administrator" permission in the server and then allow you to do a command like "warn server add" or something. 

Would be really pleased to see this in the near future

**Comment by VerstreuteSeele** [comment]
In general... moving from a role system to just checking if a user has admin or a different permission like manage.messages would make it easier to deploy this bot on multiple servers

**Comment by zacimac** [comment]
Here is an example of the config for use in multiple servers.
[GitHub Gist | Warnable - Double guild example](https://gist.github.com/zacimac/0f0f84e163b834878062ab820652cecd)

**Regarding the administrator/manage server permissions:**
I do try to make it easy to manage all through Discord and commands. However, the intention with this bot is that it should be configured once and then allow others based on the roles set to have access to those set features.

With how it is currently, it will also be difficult for the bot owner to manage who and who shouldn't have access. 
For example, inviting bots to Discord servers is really easy, and all they need is the bots Client ID (simple as copying a user's ID within Discord). If the bot owner forgets to untick the "Public Bot" in the Discord Developer Portal, anyone can add it and deploy it on their own server without the bot owner's permission.

TLDR: The Discord permissions can make it difficult for the bot owner to manage what servers should and shouldn't have access to the bot.

But this is a great idea. And can be easily done by just making some new config settings.
It'll just take some time and quite a few changes to make it work.

**Comment by zacimac** [comment]
**Regarding your time issue:**
You can configure your time that is used in the options of the `db.js` file that you are using. 
By default, it is set to `Australia/Brisbane`

**Comment by VerstreuteSeele** [comment]
Thanks for answering. 

I now changed my time in the setup however regarding the other thing; for me it would just be great if everyone with the "administrator" or "manage messages" permission could just juse it on all the servers its on without me having to add the new server all the time.  It would be really great if you could somehow do that. If you dont want to put that into the public bot I can understand that and I would also compensate you for the time you waste on my version for the bot then if we can agree on something reasonable.

**Comment by zacimac** [comment]
> 
> **Regarding the administrator/manage server permissions:**
> I do try to make it easy to manage all through Discord and commands. However, the intention with this bot is that it should be configured once and then allow others based on the roles set to have access to those set features.
> 
> With how it is currently, it will also be difficult for the bot owner to manage who and who shouldn't have access.
> For example, inviting bots to Discord servers is really easy, and all they need is the bots Client ID (simple as copying a user's ID within Discord). If the bot owner forgets to untick the "Public Bot" in the Discord Developer Portal, anyone can add it and deploy it on their own server without the bot owner's permission.
> 
> TLDR: The Discord permissions can make it difficult for the bot owner to manage what servers should and shouldn't have access to the bot.
> 
> But this is a great idea. And can be easily done by just making some new config settings.
> It'll just take some time and quite a few changes to make it work.

**Comment by zacimac** [comment]
I will get around to adding this in the future! :)

**Comment by VerstreuteSeele** [comment]
Hey. just wanted to check up on that. Anything new so far?
